angular/src/components/login.component.ts

```diff
@@ -132,7 +132,12 @@ export class LoginComponent extends CaptchaProtectedComponent implements OnInit
 
     togglePassword() {
         this.showPassword = !this.showPassword;
-        document.getElementById('masterPassword').focus();
+        const masterPassword: HTMLInputElement = <HTMLInputElement>document.getElementById('masterPassword');
+        const masterPasswordCursorLocation = masterPassword.selectionStart;
+        setTimeout(function(){ 
```

Comment:
I assume this is to avoid a `ExpressionChangedAfterItHasBeenCheckedError`? Using `setTimeout` to solve this problem seems hacky, I think we want to avoid this.

How about using the pattern [here](https://github.com/bitwarden/jslib/blob/e1c6e4973a2532b14e3187a79688e8b0a89e9d5c/angular/src/directives/autofocus.directive.ts#L26) instead? It's still more verbose than I'd like, but it's how I solved a similar problem last time, and it seems like better practice.

Reply:
Yeah I guess using `setTimeout` does seem a little hacky. Updated the code to use pattern suggested.